src/gui/qgsnewgeopackagelayerdialog.cpp

```diff
@@ -0,0 +1,515 @@
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsnewgeopackagelayerdialog.h"
+
+#include "qgis.h"
+#include "qgsapplication.h"
+#include "qgsproviderregistry.h"
+#include "qgsvectorlayer.h"
+#include "qgsmaplayerregistry.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgsgenericprojectionselector.h"
+
+#include "qgslogger.h"
+
+#include <QPushButton>
+#include <QSettings>
+#include <QLineEdit>
+#include <QMessageBox>
+#include <QFileDialog>
+#include <QLibrary>
+
+#include <ogr_api.h>
+#include <ogr_srs_api.h>
+#include <gdal_version.h>
+#include <cpl_error.h>
+#include <cpl_string.h>
+
+#if defined(GDAL_COMPUTE_VERSION) && GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(2,0,0)
+#define SUPPORT_GEOMETRY_LESS
+#define SUPPORT_CURVE_GEOMETRIES
+#define SUPPORT_INTEGER64
+#define SUPPORT_SPATIAL_INDEX
+#define SUPPORT_IDENTIFIER_DESCRIPTION
+#define SUPPORT_FIELD_WIDTH
+#endif
+
+QgsNewGeoPackageLayerDialog::QgsNewGeoPackageLayerDialog( QWidget *parent, Qt::WindowFlags fl )
+    : QDialog( parent, fl )
+    , mOkButton( nullptr )
+    , mTableNameEdited( false )
+    , mLayerIdentifierEdited( false )
+{
+  setupUi( this );
+
+  QSettings settings;
+  restoreGeometry( settings.value( "/Windows/NewGeoPackageLayer/geometry" ).toByteArray() );
+
+  mAddAttributeButton->setIcon( QgsApplication::getThemeIcon( "/mActionNewAttribute.png" ) );
+  mRemoveAttributeButton->setIcon( QgsApplication::getThemeIcon( "/mActionDeleteAttribute.png" ) );
+
+#ifdef SUPPORT_GEOMETRY_LESS
+  mGeometryTypeBox->addItem( tr( "Non spatial" ), wkbNone );
+#endif
+  mGeometryTypeBox->addItem( tr( "Point" ), wkbPoint );
+  mGeometryTypeBox->addItem( tr( "Line" ), wkbLineString );
+  mGeometryTypeBox->addItem( tr( "Polygon" ), wkbPolygon );
+  mGeometryTypeBox->addItem( tr( "Multi point" ), wkbMultiPoint );
+  mGeometryTypeBox->addItem( tr( "Multi line" ), wkbMultiLineString );
+  mGeometryTypeBox->addItem( tr( "Multi polygon" ), wkbMultiPolygon );
+#ifdef SUPPORT_CURVE_GEOMETRIES
+#if 0
+  // QGIS always create CompoundCurve and there's no real interest of having just CircularString. CompoundCurve are more useful
+  mGeometryTypeBox->addItem( tr( "Circular string" ), wkbCircularString );
+#endif
+  mGeometryTypeBox->addItem( tr( "Compound curve" ), wkbCompoundCurve );
+  mGeometryTypeBox->addItem( tr( "Curve polygon" ), wkbCurvePolygon );
+  mGeometryTypeBox->addItem( tr( "Multi curve" ), wkbMultiCurve );
+  mGeometryTypeBox->addItem( tr( "Multi surface" ), wkbMultiSurface );
+#endif
+
+#ifdef SUPPORT_GEOMETRY_LESS
+  mGeometryColumnEdit->setEnabled( false );
+  mCheckBoxCreateSpatialIndex->setEnabled( false );
+  mCrsSelector->setEnabled( false );
+#endif
+
+  mFieldTypeBox->addItem( tr( "Text data" ), "text" );
```

Comment:
Yes, tested at https://github.com/qgis/QGIS/pull/3031/commits/3ba86d48a92b7e86acb5e7090249860f9057b658#diff-5fd5538606f3c5b92693c0b20647c1d3R104 and https://github.com/qgis/QGIS/pull/3031/commits/3ba86d48a92b7e86acb5e7090249860f9057b658#diff-5fd5538606f3c5b92693c0b20647c1d3R165